juju/api.go

```diff
@@ -453,10 +362,11 @@ func PrepareEndpointsForCaching(info configstore.EnvironInfo, hostPorts [][]netw
 		preferIPv6 := maybePreferIPv6(info)
 		network.SortHostPorts(uniqueHPs, preferIPv6)
 
-		if addrConnectedTo.Value != "" {
-			return network.EnsureFirstHostPort(addrConnectedTo, uniqueHPs)
+		for _, addr := range addrConnectedTo {
+			if addr.Value != "" {
```

Comment:
Sorry, I didn't see the context properly, I thought you had added a check where there wasn't one before. Leave it, unless you're confident it won't happen (I'm not).